src/iqm/iqm_client/transpile.py

```
@@ -0,0 +1,375 @@
+
+    Args:
+        circuit (Circuit): The circuit to add Move instructions to.
+        arch (QuantumArchitectureSpecification): Restrictions of the target device
+        existing_moves (ExistingMoveHandlingOptions, optional): Specifies how to deal with existing move instruction,
+            If None, the function will use ExistingMoveHandlingOptions.REMOVE with a user warning if there are move
+            instructions in the circuit. Defaults to None.
+          qubit_mapping: Mapping of logical qubit names to physical qubit names.
+              Can be set to ``None`` if all ``circuits`` already use physical qubit names.
+    """
+    res_status = ResonatorStateTracker.from_quantum_architecture_specification(arch)
+    if not qubit_mapping:
+        qubit_mapping = {}
+    for q in arch.qubits:
+        if q not in qubit_mapping.values():
+            qubit_mapping[q] = q
+    existing_moves_in_circuit = [i for i in circuit.instructions if i.name == res_status.moveGate]
+    if not res_status.supports_move:
+        if not existing_moves_in_circuit:
+            return circuit
+        if existing_moves == ExistingMoveHandlingOptions.REMOVE:
+            return transpile_remove_moves(circuit)
+        raise ValueError('Circuit contains Move instructions, but device does not support them')
+
+    if existing_moves is None and len(existing_moves_in_circuit) > 0:
+        warnings.warn('Circuit already contains Move Instructions, removing them before transpiling.')
+        existing_moves = ExistingMoveHandlingOptions.REMOVE
+    if existing_moves is None or existing_moves == ExistingMoveHandlingOptions.REMOVE:
+        circuit = transpile_remove_moves(circuit)
+    elif existing_moves == ExistingMoveHandlingOptions.KEEP:
+        try:
+            IQMClient._validate_circuit_moves(arch, circuit, qubit_mapping=qubit_mapping)
+        except CircuitExecutionError as e:
+            if e.args[0].startswith('Instruction prx on'):
```

Is the CircuitExecutionError silently ignored if `e.args[0]` does not start with "Instruction prx on"?
...or should there be some handling for that case too?

---

It's silently ignored because that's when we want to add the move gate. If a prx is not allowed, the circuit is wrong.

---

Removed the silently ignoring after discussion with @smite   
Although we can fix the other errors by adding move gates. It is ambiguous what the user intended, so it errors out. e.g. A circuit with an odd number of moves; which move is missing?